Haskell/Blarney/Backend/Simulation.hs

```diff
@@ -54,18 +54,17 @@ simDontCare = 0
 -- | Simulate a 'Simulator' until a 'Finish' 'Prim' is triggered.
 runSim :: Simulator -- ^ the 'Simulator' to simulate, compiled with 'compileSim'
```

Comment:
Maybe add a comment that this function is strict in the sense that it will evaluate all effects before returning the outputs? (This is fine, actually it's kind of the point of this function, but it differs from `compileSim` which gives you something lazy.)